### DISCUSSION TITLE
Upgrade Buster to Bullseye - PDNS server ignored?

### DISCUSSION THREAD

**Discussion by clusterx** [PowerDNS/pdns#14007] [Q&A] [answered]
I just upgraded from Buster to Bullseye, previously I adjusted the pdns.list for PDNS server accordingly:
deb [signed-by=/etc/apt/keyrings/auth-48-pub.asc arch=amd64] http://repo.powerdns.com/debian bullseye-auth-48 main

The upgrade went smoothly, pdns server is also running and syncing the secondary NS via API ...
However, no upgrade for PDNS server (4.8.4 same as before w/ debian10) was carried out during the distro upgrade and I'm wondering if this is a bug or if there is no difference between buster and bullseye for the PDNS server?

**Comment by klaus-nicat** [comment]
If your apt configuration points to -auth-48, then you get the latest PowerDNS 4.8.x. So if you had 4.8.4 before, you get also 4.8.4.

As far as I know, Debian updates packages if there is a newer version. If the version does not change then there is no need upgrade. Sometimes, when the the distribution name is part of the verison number, it can happen that "buster" is newer than "bullseye" because of lexical ordering.

Anyways, you can check which PDNS is installed (from the bullseye or buster repo) with "apt policy pdns-server". If it does not show a repo, then it means that the installed version was installed from some repo that is not configured anymore.

**Comment by clusterx** [comment]
hmm, the new OS doesn't seem to be taken into account w/  the pdns.list possibly a bug ...

"# apt policy pdns-server"
pdns-server:
  Installed: 4.8.4-1pdns.buster
  Candidate: 4.8.4-1pdns.buster
  Version table:
 *** 4.8.4-1pdns.buster 100
        100 /var/lib/dpkg/status
     4.8.4-1pdns.bullseye 500
        500 http://repo.powerdns.com/debian bullseye-auth-48/main amd64 Packages
     4.4.1-1 500
        500 http://deb.debian.org/debian bullseye/main amd64 Packages
        500 http://mirror.hetzner.com/debian/packages bullseye/main amd64 Packages

and reinstall does not work ...
"apt --reinstall install pdns-server pdns-backend-mysql"

pdns-server is already the newest version (4.8.4-1pdns.buster).

Reinstallation of pdns-backend-mysql is not possible, it cannot be downloaded.
Reinstallation of pdns-server is not possible, it cannot be downloaded.

**Comment by mnordhoff** [comment, marked as answer]
apt doesn't really know the meaning of words in versions, it's just sorting them based on its rules about numbers and letters and other characters.

You should be able to force apt to "downgrade" from 4.8.4-1pdns.bu**ster** to 4.8.4-1pdns.bu**llseye** with a command line "`sudo apt install pdns-backend-mysql=4.8.4-1pdns.bullseye pdns-server=4.8.4-1pdns.bullseye`".

Or you could take this opportunity to switch to the 4.9 repository and upgrade to 4.9.0, since that's unambiguously a higher version number regardless of the distro codename. (As always, read the [announcement](https://blog.powerdns.com/2024/03/15/authoritative-server-4-9-0) and [upgrade notes](https://doc.powerdns.com/authoritative/upgrading.html) before upgrading.)

**Comment by clusterx** [comment]
@mnordhoff
Thank you for the downgrade explanation on this, that worked!
Yes, it was planned to update PDNS afterwards, but I wasn't sure whether the buster packages wouldn't have caused conflicts in that case.

Thank you so much guys!

pdns-server:
  Installed: 4.8.4-1pdns.bullseye
  Candidate: 4.8.4-1pdns.bullseye
